src/main/java/com/iexec/resultproxy/jwt/JwtService.java

```diff
@@ -86,10 +75,12 @@ private byte[] initKey(String jwtKeyPath) throws IOException {
      * <ul>
      * <li> If a JWT is not found in storage for the given wallet address.
      * <li> If a JWT is found but was not signed with the correct key.
+     * </ul>
+     *
      * @param walletAddress Ethereum address for which
      * @return A valid JWT token signed with this instance key.
      */
-    private String getOrCreateJwt(String walletAddress) {
+    public synchronized String getOrCreateJwt(String walletAddress) {
```

Comment:
This allows a single user (Worker) to ask for a JWT at a time. Is that what we really want? We probably want a more fine-grained control mechanism, such as a `ContextualLockRunner` - as it was done in the now-deleted `String getOrCreateJwt(SignedChallenge signedChallenge)` method.

Reply:
honestly, the result proxy only creates a single token here.
We are talking about a query per valid worker-taskid combination.
The synchronized granularity should be sufficient enough without requesting a more detailed mechanism for the moment.
I agree to keep in mind this can be modified for your solution if I'm wrong.

Reply:
That wasn't the case before your last commit. Now, you're right that we should not expect issues :)
Let's merge as is!